standards/ICRC-3/README.md

```diff
@@ -0,0 +1,206 @@
+# `ICRC-3`: Transaction Log
+
+| Status |
+|:------:|
+| Draft  |
+
+`ICRC-3` is a standard for accessing the transaction log of a Ledger on the [Internet Computer](https://internetcomputer.org).
+
+`ICRC-3` specifies:
+1. A generic format for sharing the transaction log without information loss
+2. A mechanism to verify the transaction log on the client side to allow downloading the transaction log via query calls
+3. A way for new standards to define new transaction types compatible with ICRC-3
+
+## Transaction Log
+
+The transaction log is a list of transactions where each transaction contains the hash of its parent (`phash`). The parent of a transaction `i` is transaction `i-1` for `i>0` and `null` for `i=0`.
+
+```
+   ┌──────────────────────┐          ┌──────────────────────┐
+   |         Tx i         |          |         Tx i+1       |
+   ├──────────────────────┤          ├──────────────────────┤
+◄──| phash = hash(Tx i-1) |◄─────────| phash = hash(Tx i)   |
+   | ...                  |          | ...                  |
+   └──────────────────────┘          └──────────────────────┘
+
+```
+
+## Value
+
+The [candid](https://github.com/dfinity/candid) format supports sharing information even when the client and the server involved do not have the same schema (see the [Upgrading and subtyping](https://github.com/dfinity/candid/blob/master/spec/Candid.md#upgrading-and-subtyping) section of the candid spec). While this mechanism allows to evolve services and clients
+independently without breaking them, it also means that a client may not receive all the information that the server is sending, e.g. in case the client schema lacks some fields that the server schema has.
+
+This loss of information is not an option for `ICRC-3`. The client must receive the same exact data the server sent. For this reason, `ICRC-3` introduces the `Value` type which never changes: 
```

Comment:
Where?

Reply:
> The [candid](https://github.com/dfinity/candid) format supports sharing information even when the client and the server involved do not have the same schema (see the [Upgrading and subtyping](https://github.com/dfinity/candid/blob/master/spec/Candid.md#upgrading-and-subtyping) section of the candid spec). While this mechanism allows to evolve services and clients
> independently without breaking them, it also means that a client may not receive all the information that the server is sending, e.g. in case the client schema lacks some fields that the server schema has.
>
> This loss of information is not an option for `ICRC-3`. The client must receive the same exact data the server sent.

Reply:
Yeah, that's the text that I commented on. It simply states that it's not acceptable, but it does not explain why.

Reply:
I've added the following:

> This loss of information is not an option for `ICRC-3`. The client must receive the same exact data the server sent in order to verify it. Verification is done by hashing the data and checking that the result is consistent with what has been certified by the server.

Is it better?

Reply:
Thanks. Why isn't it up to the client if it wants to verify the hashes? If it wants to verify the hashes, it should make sure to have an up-to-date schema. If it only wants certain data, it can use any schema that provides that data. By verifying the hash it can even make sure that its schema is up-to-date so.

Reply:
The client cannot check that the schema is identical to the one of the canister. There is no guarantee that the schema will be identical even if the client first fetches it from the canister. Candid was built with a different goal in mind: the schema of the server can evolve independently from the client.
The solution to this problem is to have a type that will never change over time so it is guaranteed that clients and canisters will always know the full structure. That's what `Value` is.

Reply:
> If it wants to verify the hashes, it should make sure to have an up-to-date schema.

Because it's not a sustainable solution in a decentralized system. If we went this way, releasing a new Ledger feature would immediately break block validation for all existing clients. We don't have a full list of our clients, so we cannot ask them to upgrade in advance.
The main clients of ICRC-3 standard are the on-chain and off-chain indexers, such as the index canister and Rosetta. We want these to continue working correctly with no intervention as we evolve the ledger.